### DISCUSSION TITLE
change FEC for comparison

### DISCUSSION THREAD

**Discussion by WuShengHan** [NVlabs/sionna#152] [General]
Hello everyone,

I would like to express my gratitude for the support in the Sionna project. Currently, I am conducting research on next-gen FEC, specifically focusing on LDPC design and comparison.

One significant challenge in FEC is the presence of an error floor, which makes it impractical to perform extensive and super-long testing over tarabyte data. Therefore, I propose utilizing hard decoding as a fair basis for comparison.

I would appreciate insights on how to modify the LDPC parity matrix in the Sionna. I aim to compare its performance with BG1, BG2, and Polar Codes. is this easy to modify LDPC parity matrix and able to switch to hard decoding for comparison? 

Thank you all in advance for your guidance and expertise. I am looking forward to your valuable input.
My vedio about FEC promotion: https://youtu.be/q0nSklM64rU

**Comment by SebastianCa** [comment]
Hello WuShengHan,

The ``sionna.fec.ldpc.decoding.LDPCBPDecoder`` can operate on any parity-check matrices You can learn more about its functionalities from the corresponding [API documentation](https://nvlabs.github.io/sionna/api/fec.ldpc.html#sionna.fec.ldpc.decoding.LDPCBPDecoder).

For a practical example, you may wish to refer to our [Weighted BP tutorial notebook](https://nvlabs.github.io/sionna/examples/Weighted_BP_Algorithm.html). It provides a clear guide on how to load the parity-check matrix of a BCH code.

Regarding a hard-decision decoding, while there is no dedicated hard-decision decoder avaiable, you could try to use the min-sum decoder for your scenario. It should work adequately if you set the input LLRs to +/- `llr_max`, where `llr_max` represents a constant.
